api/lib/models/checkout-items/checkout-items-data-mapper-impl.ts

```diff
@@ -68,13 +60,90 @@ export class CheckoutItemsDataMapperImpl extends GenericDataMapper
     ).toPromise();
   }
 
-  public getAllAvailable(): Promise<IDbResult<Stream<CheckoutItems>>> {
+  public get(id: UidType, opts?: IUowOpts): Promise<IDbResult<CheckoutItems>> {
+    let queryBuilder = squel.select({ 
+      autoQuoteFieldNames: true, 
+      autoQuoteTableNames: true 
+    })
+      .from(this.tableName);
+    if (opts && opts.fields) {
+      queryBuilder = queryBuilder.fields(opts.fields);
+    }
+    queryBuilder = queryBuilder
+      .where(`${this.pkColumnName}= ?`, id);
+    const query = queryBuilder.toParam();
+    query.text = query.text.concat(';');
+    return from(this.sql.query<CheckoutItems>(query.text, query.values, { stream: false, cache: true }))
+      .pipe(
+        map((checkoutItems: CheckoutItems) => ({ result: 'Success', data: checkoutItems })),
+      )
+      .toPromise();
+  }
+
+  public async getAll(opts?: IUowOpts): Promise<IDbResult<Stream<CheckoutItems>>> {
+    let queryBuilder = squel.select({
+      autoQuoteFieldNames: true,
+      autoQuoteTableNames: true,
+    })
+      .from(this.tableName, 'checkoutItems');
+      if(opts && opts.fields) {
+        queryBuilder = queryBuilder.fields(opts.fields);
+      }
+      if(opts && opts.startAt) {
+        queryBuilder = queryBuilder.offset(opts.startAt);
+      }
+      if(opts && opts.count) {
+        queryBuilder = queryBuilder.limit(opts.count);
+      }
+      if (opts && opts.byHackathon) {
+        queryBuilder = queryBuilder.
+            where(
+              'hackathon_id = ?',
+              await (opts.hackathon ?
+                Promise.resolve(opts.hackathon) :
+                this.activeHackathonDataMapper.activeHackathon.pipe(map(hackathon => hackathon.uid)).toPromise()),
+            );
+      }
+      const query = queryBuilder
+        .toString()
```

Comment:
Since this query takes `?` parameters, use the `toParam()` instead of `toString()`. In all honesty it is probably best to use toParam just in case for security